### PR TITLE
[PublishBuildArtifactsV1] Delete hardcoded "/E" option not to copy subdirectories

### PR DESCRIPTION
**Pull request by tatyana-kostromskaya**
**Task name**: PublishBuildArtifactsV1

**Description**: 
Delete from robocopy call hardcoded "/E" option and replace it with variable. This variable will contain "/E" option if "PathToPublish" is not a file, and it will be empty string in case "PathToPublish" is file

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) #14984 

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Comment by max-zaytsev** [comment]
@tatyana-kostromskaya, @anatolybolshakov  Could this be a breaking change for the customers who use this task with a current behavior?

**Comment by tatyana-kostromskaya** [comment]
> @tatyana-kostromskaya, @anatolybolshakov Could this be a breaking change for the customers who use this task with a current behavior?

I discussed it with @anatolybolshakov. This behavior is not documented so we can consider these changes as bug fixing of unexpected behavior.

**Review by max-zaytsev — APPROVED** [review]

File: Tasks/PublishBuildArtifactsV1/Invoke-Robocopy.ps1 [hunk 1]
```diff
@@ -85,7 +86,7 @@ if ($ParallelCount -gt 1) {
     }
 }
 else {
-    & robocopy.exe /E /COPY:DA /NP /R:3 $Source $Target $File 2>&1 |
+    & robocopy.exe $CopySubFoldersOption /COPY:DA /NP /R:3 $Source $Target $File 2>&1 |
```
Comment (anatolybolshakov) [inline Tasks/PublishBuildArtifactsV1/Invoke-Robocopy.ps1:89]: Could you please add some unit tests to cover it?

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks! Could you please take a look at the comments?

File: Tasks/PublishBuildArtifactsV1/Invoke-Robocopy.ps1 (hunk 1 above)
Comment (tatyana-kostromskaya) [reply Tasks/PublishBuildArtifactsV1/Invoke-Robocopy.ps1:89]: We don't have tests that covers this ps1 file. And I'm not sure how this function can be tested. Mostly this script contains invocation of robocopy.exe and I an not sure how I can mock it

Comment (anatolybolshakov) [reply Tasks/PublishBuildArtifactsV1/Invoke-Robocopy.ps1:89]: Ok, this sounds like a more wide task we could pick up later

**Review by anatolybolshakov — APPROVED** [review]

**Pull request merged**